byway/src/main/java/com/google/sps/data/UserInfo.java

```diff
@@ -58,6 +62,10 @@ public String getEmail() {
     return Collections.unmodifiableList(this.tripIds);
   }
 
+  public void addTripId(String tripKey) {
+    tripIds.add(tripKey);
+  }
```

Comment:
Consider taking the `Key` as the argument instead of the string form of the key. That way you can do `checkArgument(tripKey.getKind().equals(Trip.DATASTORE_ENTITY_KIND))`, which prevents bugs by checking that the entity kind matches (similar to in the `fromEntity` method).

For even more type-safety, you could have a method `addTrip(Trip trip)` and just implement it like:
```java
void addTrip(Trip trip) {
  // Trip class is trusted to provide a valid value for its `keyString`, so you don't need to do any extra checking;
  // the fact that the argument is definitely a `Trip` is sufficient to know this is the case (this is why we in practice
  // often use custom types for things that *could* just be represented as a string or list; the custom type promises
  // to maintain some invariant, and therefore we can avoid extra checks, because the compile-time type check is
  // sufficient to prove that the invariant holds).
  tripIds.add(trip.getKeyString());
}
```

Reply:
done